lib/jormungandr/test/integration/Test/Integration/Jormungandr/Scenario/API/StakePools.hs

```diff
@@ -493,6 +503,52 @@ spec = do
         r <- joinStakePool ctx (p ^. #id) (w, "Secure Passprase")
         expectResponseCode HTTP.status404 r
 
+    it "STAKE_POOLS_ESTIMATE_FEE_01 - fee matches eventual cost" $ \ctx -> do
+        (_, p:_) <- eventually $
+            unsafeRequest @[ApiStakePool] ctx listStakePoolsEp Empty
+        w <- fixtureWallet ctx
+        fee <- getFromResponse amount <$> joinStakePoolFee ctx (p ^. #id) w
+        r <- joinStakePool ctx (p ^. #id) (w, fixturePassphrase)
+        verify r
+            [ expectFieldEqual amount fee
+            ]
```

Comment:
**Question**: Is this expectation valid?

I was under the impression that the precision of fee estimation cannot be guaranteed, because of randomization that occurs during coin selection. See: https://github.com/input-output-hk/cardano-wallet/issues/1094#issuecomment-559773456

If there is randomization going on under the hood, then will this occasionally trigger a test failure?

Reply:
This is true only by "accident". The wallet test UTxO are all very homogeneous (10 coins of the same amount). So, this compensate the random aspect of it. 

Yet, I agree that checking this in a scenario is misleading and somewhat wrong. Especially